Sources/Apodini/Metadata/MetadataWithDelegatingHandler.swift

```diff
@@ -0,0 +1,65 @@
+//
+// Created by Andreas Bauer on 02.07.21.
+//
+
+/// A type erased `DefinitionWithDelegatingHandler`.
+public protocol AnyDefinitionWithDynamicDelegatingHandler {
+    /// Internal method used to add the `DelegatingHandlerInitializer` provided
+    /// by the `DefinitionWithDelegatingHandler` to the `ContextNode`.
+    ///
+    /// Note, this method does nothing, if self also conforms to `DefinitionWithDelegatingHandlerKey`.
+    /// - Parameter visitor: The `SyntaxTreeVisitor`.
+    func addInitializerContextValue(_ visitor: SyntaxTreeVisitor)
```

Comment:
If I get this correctly, this is just for metadata definitions that want to manipulate both, some other ContextKey and the DelegatingHandlerContextKey? If so, wouldn't it be better to provide a more generic way for a metadata definition to manipulate any number of different ContextKeys? Also, maybe it would be better to prefix this protocol with an underscore if it can't be omitted or made internal.

Reply:
I have already thought about this as well. I don't think there is a good way of doing that elegantly.
Of course there is always the way of overriding the `accept` yourself. So its not impossible to do it, its just that it remove a lot of convenience if you have to always implement the `accept` method yourself. And as Metadata will typically contribute exactly one value (or e.g. merge multiple values in a struct) I chose to make that typical use case convenient, but the untypical use case is still not impossible.

But feel free to leave any suggestions.

Reply:
Maybe with our new idea of having a stack of delegating Handlers (mentioned [here](https://github.com/Apodini/Apodini/pull/310#discussion_r664470138)), the MetadataBlock would indeed be the best solution. Otherwise, I'd be fine with leaving it as is, since it is generic to `DelegatingHandlerInitializer`s and we don't have discovered this requirement for other Metadata yet.

Reply:
I think the DelegatingHandlerInitializer Metadata are special in the way that you may need to pass in some initializer arguments, which wouldn't work that easy when using a `MetadataBlock`.
Therefore I would leave those protocols in for now. Maybe we can remove one of those in the future again I we see that we don't really need it.